LibreHardwareMonitorLib/Hardware/RyzenSMU.cs

```diff
@@ -0,0 +1,639 @@
+﻿// ported from: https://gitlab.com/leogx9r/ryzen_smu
+// and: https://github.com/irusanov/SMUDebugTool
+
+using LibreHardwareMonitor.Hardware.CPU;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace LibreHardwareMonitor.Hardware
+{
+    internal class RyzenSMU
+    {
+        private Mutex _mutex = new Mutex();
+        private uint _dram_base_addr = 0;
+        private uint _pm_table_version = 0;
+        private uint _pm_table_size = 0;
+        private uint _pm_table_size_alt = 0;
+        private bool _supported_cpu = false;
+
+        private const uint SMU_RETRIES_MAX = 8096;
+        private const byte SMU_PCI_ADDR_REG = 0xC4;
+        private const byte SMU_PCI_DATA_REG = 0xC8;
+        private const uint SMU_REQ_MAX_ARGS = 6;
+
+        private CpuCodeName _cpu_code_name;
+        private uint _cmd_addr;
+        private uint _rsp_addr;
+        private uint _args_addr;
+
+        public RyzenSMU(uint family, uint model, uint packageType)
+        {
+            _cpu_code_name = GetCpuCodeName(family, model, packageType);
+
+            _supported_cpu = SetAddrs(_cpu_code_name);
+
+            SetupPmTableAddrAndSize();
+        }
+
+        private CpuCodeName GetCpuCodeName(uint family, uint model, uint packageType)
+        {
+            if (family == 0x17)
+            {
+                switch (model)
+                {
+                    case 0x01:
+                        if (packageType == 7)
+                            return CpuCodeName.Threadripper;
+                        else
+                            return CpuCodeName.SummitRidge;
+                    case 0x08:
+                        if (packageType == 7)
+                            return CpuCodeName.Colfax;
+                        else
+                            return CpuCodeName.PinnacleRidge;
+                    case 0x11:
+                        return CpuCodeName.RavenRidge;
+                    case 0x18:
+                        if (packageType == 2)
+                            return CpuCodeName.RavenRidge2;
+                        else
+                            return CpuCodeName.Picasso;
+                    case 0x20:
+                        return CpuCodeName.Dali;
+                    case 0x31:
+                        return CpuCodeName.CastlePeak;
+                    case 0x60:
+                        return CpuCodeName.Renoir;
+                    case 0x71:
+                        return CpuCodeName.Matisse;
+                    case 0x90:
+                        return CpuCodeName.Vangogh;
+                    default:
+                        return CpuCodeName.Undefined;
```

Comment:
Braces in multi line switches to follow code style.

Reply:
Sure, fixed